specification/postgresql/resource-manager/Microsoft.DBforPostgreSQL/preview/2024-11-01-preview/TuningOptions.json

```diff
@@ -0,0 +1,456 @@
+              "description": "Workload start time in UTC date-time string format."
+            },
+            "endTime": {
+              "type": "string",
+              "format": "date-time",
+              "description": "Workload end time in UTC date-time string format."
+            },
+            "queryCount": {
+              "format": "int32",
+              "type": "integer",
+              "description": "Workload query examined count. For DROP INDEX will be 0."
+            }
+          },
+          "description": "Stores workload information for the recommended action."
+        },
+        "estimatedImpact": {
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/ImpactRecord"
+          },
+          "x-ms-identifiers": [
+            "dimensionName"
+          ],
+          "readOnly": true,
+          "description": "The estimated impact of this recommended action"
+        },
+        "details": {
+          "$ref": "#/definitions/IndexRecommendationDetails",
+          "description": "Stores recommendation details for the recommended action."
+        }
+      },
+      "description": "Index recommendation properties."
+    },
+    "IndexRecommendationResource": {
+      "type": "object",
+      "properties": {
+        "type": {
+          "description": "The type of resource.",
+          "type": "string",
+          "readOnly": true
+        },
+        "kind": {
+          "description": "Kind of index recommendation.",
+          "type": "string",
+          "readOnly": true
+        },
+        "id": {
+          "type": "string",
+          "description": "The id of index recommendation.",
+          "readOnly": true
+        },
+        "name": {
+          "type": "string",
+          "description": "The name of index recommendation.",
+          "readOnly": true
+        },
+        "properties": {
+          "$ref": "#/definitions/IndexRecommendationResourceProperties",
+          "x-ms-client-flatten": true,
+          "description": "Properties of IndexRecommendationResource."
+        }
+      },
+      "description": "Index recommendation properties."
+    },
+    "IndexRecommendationListResult": {
+      "description": "A list of available index recommendations.",
+      "type": "object",
+      "properties": {
+        "value": {
+          "description": "Collection of available index recommendations.",
+          "uniqueItems": false,
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/IndexRecommendationResource"
+          },
+          "x-ms-identifiers": [
+            "name"
+          ]
+        },
+        "nextLink": {
+          "description": "URL client should use to fetch the next page (per server side paging).\r\nIt's null for now, added for future use.",
+          "type": "string"
+        }
+      }
+    },
+    "TuningOptionsResource": {
+      "type": "object",
+      "properties": {
+        "id": {
```

Comment:
>id [](http://example.com/codeflow?start=9&length=2)

Get top-level properties (id, type, and name) from common-types definitions/Resource (or ProxyResource).